internal/agent/device/bootstrap.go

```diff
@@ -0,0 +1,295 @@
+		}
+	}
+	if !approved {
+		b.log.Infof("%senrollment request not yet approved", b.logPrefix)
+		return false, nil
+	}
+	if len(*enrollmentRequest.Status.Certificate) == 0 {
+		b.log.Infof("%senrollment request approved, but certificate not yet issued", b.logPrefix)
+		return false, nil
+	}
+	b.log.Infof("%senrollment approved and certificate issued", b.logPrefix)
+
+	if _, err = cert.ParseCertsPEM([]byte(*enrollmentRequest.Status.Certificate)); err != nil {
+		return false, fmt.Errorf("parsing signed certificate: %v", err)
+	}
+
+	b.log.Infof("%swriting signed certificate to %s", b.logPrefix, b.managementGeneratedCertFile)
+	if err := b.deviceWriter.WriteFile(b.managementGeneratedCertFile, []byte(*enrollmentRequest.Status.Certificate), os.FileMode(0600)); err != nil {
+		return false, fmt.Errorf("writing signed certificate: %v", err)
+	}
+
+	return true, nil
+}
+
+func (b *Bootstrap) writeEnrollmentBanner() error {
+	if b.enrollmentUIEndpoint == "" {
+		b.log.Warningf("%sflightctl enrollment UI endpoint is missing, skipping enrollment banner", b.logPrefix)
+		return nil
+	}
+	url := fmt.Sprintf("%s/enroll/%s", b.enrollmentUIEndpoint, b.deviceName)
+	if err := b.writeQRBanner("\nEnroll your device to flightctl by scanning\nthe above QR code or following this URL:\n%s\n\n", url); err != nil {
+		return fmt.Errorf("failed to write device enrollment banner: %w", err)
+	}
+	return nil
+}
+
+func (b *Bootstrap) writeManagementBanner() error {
+	// write a banner that explains that the device is enrolled
+	if b.enrollmentUIEndpoint == "" {
+		b.log.Warningf("%sflightctl enrollment UI endpoint is missing, skipping management banner", b.logPrefix)
+		return nil
+	}
+	url := fmt.Sprintf("%s/manage/%s", b.enrollmentUIEndpoint, b.deviceName)
+	if err := b.writeQRBanner("\nYour device is enrolled to flightctl,\nyou can manage your device scanning the above QR. or following this URL:\n%s\n\n", url); err != nil {
+		return fmt.Errorf("%sfailed to write device management banner: %w", b.logPrefix, err)
+	}
+	return nil
+}
+
+func (b *Bootstrap) writeQRBanner(message, url string) error {
+	qrCode, err := qrcode.New(url, qrcode.High)
+	if err != nil {
+		return fmt.Errorf("failed to generate new QR code: %w", err)
+	}
+
+	// convert the QR code to a string.
+	qrString := qrCode.ToSmallString(false)
+
+	// write a banner that explains that the device is enrolled
+	buffer := bytes.NewBufferString("\n")
+	buffer.WriteString(qrString)
+
+	// write the QR code to the buffer
+	fmt.Fprintf(buffer, message, url)
+
+	// duplicate file to /etc/issue.d/flightctl-banner.issue
+	if err := b.deviceWriter.WriteFile("/etc/issue.d/flightctl-banner.issue", buffer.Bytes(), os.FileMode(0666)); err != nil {
+		return fmt.Errorf("failed to write banner to disk: %w", err)
+	}
+
+	if err := SdNotify("READY=1"); err != nil {
+		b.log.Warningf("failed to notify systemd: %v", err)
+	}
+
+	// additionally print the banner into the output console
+	b.log.Info(buffer.String())
```

Comment:
We need to print this through plain stdout, otherwise logging breaks format:

<img width="1041" alt="image" src="https://github.com/flightctl/flightctl/assets/1176434/38d47473-f430-4bfa-a283-c96dc80b9786">